instagram-app/src/App.js

```diff
@@ -1,52 +1,17 @@
 import React from "react";
-import "./App.css";
-import dummyData from "./dummy-data";
-import PostContainer from "./components/PostContainer/PostContainer";
-import SearchBar from "./components/SearchBar/SearchBar"
+import CSS from "./App.css"
+import PostsPage from "./components/PostContainer/PostsPage"
+import withAuthenticate from "./components/authentication/withAuthenticate"
+import Login from "./components/Login/Login";
 
-class App extends React.Component {
-  state = { 
-    data: [],
-    filteredPosts: []
-  };
- 
-//BOOM! No more constructor and super
-// Why? Because syntactic sugar in new react - constructor props + super props
-//is just assumed!
-
-  componentDidMount() {
-    this.setState({ data: dummyData });
-  }
-
-  searchfilter = e => {
-    const filtered = this.state.data.filter(post => post.username.toLowerCase().includes(e.target.value.toLowerCase()))
-//check if username is equal to value, toLowercase is just in case of aberations innames
-  this.setState({ filteredPosts: filtered})
-    //when a hit results, send it to the state above, filteredPosts
-  }
-
- 
-  // const newFilter = this.state.data.filter(post => 
-    //   post.username === this.state.search
-    // )
-  //wanna filter our data ^^
-  //take this.state.data, and instead of mapping it, FILTER it
-  //go over each surface level object, assign it to post, and check if - 
-  //filter posts, only show usersnames equal to the search
-  //remember filter gets RID of things, opposite of map
+const ComponentFromWithAuthenticate = withAuthenticate(PostsPage)(Login)
 
+class App extends React.Component {
+  
   render() {
-    // maps thru dummyData, sends each object (3) to PostContainer as
-    // its own set of data
-
     return (
       <div className="App">
-        <SearchBar 
-          newSearch={this.state.search}
-          searchFilter={this.searchFilter}
-        />
-
-        <PostContainer data={this.state.data} filteredPosts={this.state.filteredPosts} searchFilter={this.searchFilter} />
+        <ComponentFromWithAuthenticate />
```

Comment:
Great job here storing the actual component call like this, another method you could use wrap your app in the HOC on the Export.Line 20. It would look like 

```js
export default withAuthenticate(PostsPage)(LoginPage);
```